src/domain/common/entities/Period.ts

```diff
@@ -0,0 +1,25 @@
+export type Period = {
+    id: string;
+    label: string;
+};
+
+export enum PeriodType {
+    DAILY = "Daily",
+    WEEKLY = "Weekly",
+    MONTHLY = "Monthly",
+    QUARTERLY = "Quarterly",
+    YEARLY = "Yearly",
+}
+
+export function validatePeriodType(periodType: string): PeriodType {
+    switch (periodType) {
+        case PeriodType.DAILY:
+        case PeriodType.WEEKLY:
+        case PeriodType.MONTHLY:
+        case PeriodType.QUARTERLY:
+        case PeriodType.YEARLY:
+            return periodType;
+        default:
+            throw new Error(`Invalid period type: ${periodType}`);
```

Comment:
I think this could be considered a breaking change, because existing dataSets with other periodTypes (like FinancialApril) work for `viewTypes` where we don't generate multiple periods (like table or grid) and with this change we're throwing this error. Can we add an "unknown" `periodType` instead of throwing the error? 

I don't remember other clients using a different period type than "YEARLY", but just to be extra safe.

Does this make sense?